### ISSUE TITLE
Request for option to be placed vertically rather than horizontally

### ISSUE THREAD

**Issue by fossfreedom** [spheras/desktopfolder#213]
Request from the UB community

> Are any options to sort by column rather than line planned ? Or ability to transform a folder-on-desktop into a panel-on-desktop ?

Think this has been raised through the observation that Files (Nautilus) arranges its icons vertically rather than horizontally.  So this is an enhancement request for a vertical arrangement option.

**Comment by aljelly** [comment]
Probably this is the correct function to edit:

https://github.com/spheras/desktopfolder/blob/839ffe1783903c788d0a060d48ce9efd09caf0a3/src/logic/folderarrangement/FolderArrangement.vala#L112

From a very quick glance it might be as simple as giving the sorting logic generic variables (like `direction` = x/y and `direction_boundary` = width/height or something) and then switching the generic variables and a few other variables between them depending on a gschema enum.

Maybe the direction should also be mirrored for RTL layouts (as in LTR -> RTL). What do other desktops do?

**Comment by fossfreedom** [comment]
Indeed - my understanding is that Files/Nautilus has open issues about RTL handling

So yes in theory  - icons should flow right to left - or with this option, from the right top-to-bottom.  In practise it seems much harder to implement & catch issues for someone who isn't a native RTL user.  Think RTL should be a separate issue.

**Comment by spheras** [comment]
mm? sounds strange to me. The problem I see is the scroll. For me, the natural growth of items is from top to bottom, in a way that if the item's don't fit in the "panel" the scroll appears vertically. When moving the central wheel of the mouse, the vertical scrolls works perfectly showing all the content of the panel.

The other way, icons will grow to the right, a horizontal scroll will appear.. Is that what people want?? or I'm missing something?

**Comment by fossfreedom** [comment]
I suspect this is an expectations point-of-view - ever since Nautilus has existed (2001?) icons are arranged  vertically - not horizontally.

**Comment by spheras** [comment]
sorry, I feel I'm lost in translation :rofl:  This is a video using nautilus in UB, I don't see differences how desktopfolder arrange icons... 
![nautilus](https://user-images.githubusercontent.com/3862933/49414760-c64cf780-f774-11e8-8d2e-2bc1d571d75d.gif)

what am I missing?

**Comment by fossfreedom** [comment]
Ah. Right click and choose to arrange the icons.

This will arrange vertically.

On Tue, 4 Dec 2018, 02:31 José Amuedo Salmerón <notifications@github.com<mailto:notifications@github.com> wrote:

sorry, I feel I'm lost in translation 🤣 This is a video using nautilus in UB, I don't see differences how desktopfolder arrange icons...
[nautilus]<https://user-images.githubusercontent.com/3862933/49414760-c64cf780-f774-11e8-8d2e-2bc1d571d75d.gif>

what am I missing?

**Comment by fossfreedom** [comment]
![image](https://user-images.githubusercontent.com/996240/49431339-2fe4fa00-f7a5-11e8-89b2-e428cdcc28c1.png)

Picture works better here! - its the "Organise Desktop by Name" option - as you can see the icons arrange vertically on the desktop

**Comment by spheras** [comment]
ahh, ok, I see.. so, that alignment only applies for the desktop icons...   If I'm not wrong, this issue is to organize vertically **desktop** icons... what about panels?  (in fact, desktop is a transparente panel, but we could provide easily a vertical alignment).. I don't see the vertical alignment for panels too. Whats your opinion?

**Comment by fossfreedom** [comment]
Correct - desktop icons.  Sorry to cause confusion.

I would suggest such an enhancement applies to both the main desktop icons transparent panel and the other visible "Panels" i.e. consistent arrangement either vertically or horizontally.

**Comment by spheras** [comment]
I can understand and share the case for the desktop: is something expected due to the experience from other desktops with icons (even windows if I don't remember bad was like that)... In the case of panels, to be honest, I don't see it. The main problem for that arrangement in panels is the scroll, icons (the grid) will grow to the right, the horizontal scroll will appear instead of the vertical.. very strange situation, don't you agree?   In my opinion, panels are like the video I posted here with a nautilus window and files inside.

**Comment by fossfreedom** [comment]
fair enough argument :)

agree

On Wed, 5 Dec 2018 at 14:41, José Amuedo Salmerón <notifications@github.com<mailto:notifications@github.com>> wrote:

I can understand and share the case for the desktop: is something expected due to the experience from other desktops with icons (even windows if I don't remember bad was like that)... In the case of panels, to be honest, I don't see it. The main problem for that arrangement in panels is the scroll, icons (the grid) will grow to the right, the horizontal scroll will appear instead of the vertical.. very strange situation, don't you agree? In my opinion, panels are like the video I posted here with a nautilus window and files inside.

**Comment by spheras** [comment]
Ok :D great, thanks for the comprehension. Anyway, I think this is easy to modify, we could put in the popup menu, below the sortby->reverse order, a similar option "vertical alignment". Vertical Alignment will be checked for the desktop and unchecked for the panels, but users can modify. I think that's the best option.

**Comment by fossfreedom** [comment]
Had a very quick look.  Seems to work nicely with existing icons on the desktop itself.

I tried a mass copy of 20 desktop files from /usr/share/applications via a terminal to ~/Desktop.

I note two observations - the icons didn't appear on the desktop until I clicked on the background of the desktop.  Focus issue?

Also the 20 icons were spread horizontally across the desktop even though my sort was "Vertical".  I unclick and click the vertical sort and everything sorted correctly.... actually with the animation the mass movement from horizontal to vertical looked superb!
